### PR TITLE
Adding documentation with doxygen for physics lib

### PR DESCRIPTION
**Pull request by SomeOne768**

### CODE REVIEW

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.hpp
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.hpp:1]: Misses File documentation

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.hpp [hunk 1]
```diff
@@ -8,27 +8,100 @@ namespace sne
...
             Vector3 _center;
 
         public:
-            // TO UPDATE
+            /**
+             * @brief Construct a new Box Collider object
+             * 
+             * @param center 
+             * @param width 
+             * @param depth
```
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.hpp]: Might be a godd idea to associate those 3 parameters with an axis (x y or z)

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.hpp [hunk 2]
```diff
@@ -8,27 +8,100 @@ namespace sne
...
+            /**
+             * @brief Destroy the Box Collider object
+             * 
+             */
             ~BoxCollider() = default;
-            const Vector3 &operator[](int) const;
+
+            /**
+            * 
+            * Create multicolor cube of size 1*1*1.
```
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.hpp]: hmm

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/SAT.hpp
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/SAT.hpp:1]: Misses file documentation

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/SphereCollider.hpp
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/SphereCollider.hpp:1]: misses files documentation

**Review by antonindevidal — APPROVED** [review]

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.hpp (hunk 1 above)
Comment (SomeOne768) [reply Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.hpp]: That's a good idea. Indeed, idk how but I really thought about like how am I supposed to describe it the best way, and thought about it geometrically, in my head. Sad to see I wasn't smart enough to use those basic letters :smiling_face_with_tear:

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.hpp (hunk 2 above)
Comment (SomeOne768) [reply Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.hpp]: Hmm hmm hmmm :eyes:

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.hpp [hunk 3]
```diff
@@ -2,33 +2,112 @@
 #include "architecture/Component.hpp"
 #include "architecture/Vector3.hpp"
 #include <vector>
-
+/**
```
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.hpp:2]: Put that before the include

**Review by antonindevidal — APPROVED** [review]

**Pull request merged**